skada/deep/base.py

```diff
@@ -469,23 +469,34 @@ def forward(
                 return self.base_module_(X, sample_weight=sample_weight)
 
 
-class DomainAwareNet(NeuralNetClassifier, _DAMetadataRequesterMixin):
+class _DomainAwareNet(NeuralNet, _DAMetadataRequesterMixin):
     __metadata_request__fit = {"sample_weight": True}
-    __metadata_request__score = {'sample_weight': True, 'sample_domain': True, 'allow_source': True}
+    __metadata_request__score = {
+        'sample_weight': True,
+        'sample_domain': True,
+        'allow_source': True
+    }
     """
-    A domain-aware neural network classifier with sample weight support.
+    Base class for a domain-aware neural network with sample weight support.
 
-    This class extends NeuralNetClassifier to handle domain-specific input data
+    This class extends NeuralNet to handle domain-specific input data
     and sample weights. It supports various input formats and provides methods
     for training, prediction, and feature extraction while considering domain
     information and sample weights.
 
+    This class should hardly, if ever, be used directly. Instead, use
+    DomainAwareNetClassifier, DomainAwareNetDualClassifier or
```

Comment:
you should remove `DomainAwareNetDualClassifier`. Let's first focus on classification and regression (and not binary classification)